.github/workflows/generate-changelog.yaml

```diff
@@ -0,0 +1,127 @@
+name: Generate changelog
+# Reusable workflow to generate changelog after release
+
+on:
+  workflow_call:
+    inputs:
+      old-tag:
+        description: Version to be replaced
+        required: true
+        type: string
+      new-tag:
+        description: Version after bump
+        required: true
+        type: string
+      gh-changelog:
+        description: Path to the Changelog.md file
+        required: true
+        type: string
+      doc-changelog:
+        description: Path to the documentation changelog (if any exists)
+        required: false
+        default: ""
+        type: string
+      config:
+        description: Path to the changelog config JSON file
+        required: false
+        default: "./.github/changelog-config.json"
+        type: string
+
+    secrets:
+      github-username:
+        description: "The GitHub username for committing the changes."
+        required: true
+      github-email:
+        description: "The GitHub email for committing the changes."
+        required: true
+      github-token:
+        description: "The GitHub token for committing the changes."
+        required: true
+
+jobs:
+  read-config-file:
+    name: Read config file
+    runs-on: ubuntu-22.04
+    outputs:
+      content: ${{ steps.read-config.outputs.content }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.repository.default_branch }}
+          # required for getting all tags
+          fetch-depth: 0
+          # required for bypassing status checks when pushing
+          token: "${{ secrets.github-token }}"
+
+      - name: Read changelog config
+        id: read-config
+        # https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#multiline-strings
+        run: |
+          content=$(cat "${{ inputs.config }}")
+          { echo "content<<EOF"; echo "$content"; echo "EOF"; } >> "$GITHUB_OUTPUT"
+        shell: bash
```

Comment:
This can be be part of `generate-changelog`, no need for a separate job just to read a file from the repository